backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionModule.java

```diff
@@ -22,39 +22,16 @@
 import static com.datastax.driver.core.DataType.cint;
 import static com.datastax.driver.core.DataType.timeuuid;
 
-import java.util.List;
-
 import org.apache.james.backends.cassandra.components.CassandraModule;
-import org.apache.james.backends.cassandra.components.CassandraTable;
-import org.apache.james.backends.cassandra.components.CassandraType;
 import org.apache.james.backends.cassandra.versions.table.CassandraSchemaVersionTable;
 
-import com.datastax.driver.core.schemabuilder.SchemaBuilder;
-import com.google.common.collect.ImmutableList;
-
-public class CassandraSchemaVersionModule implements CassandraModule {
-
-    private final List<CassandraTable> tables;
-
-    public CassandraSchemaVersionModule() {
-        this.tables = ImmutableList.of(
-            new CassandraTable(CassandraSchemaVersionTable.TABLE_NAME,
-                SchemaBuilder.createTable(CassandraSchemaVersionTable.TABLE_NAME)
-                    .ifNotExists()
-                    .addPartitionKey(CassandraSchemaVersionTable.KEY, timeuuid())
-                    .addClusteringColumn(CassandraSchemaVersionTable.VALUE, cint())
-                    .withOptions()
-                    .comment("Holds the history of the versions of the schema used.")));
-    }
-
-
-    @Override
-    public List<CassandraTable> moduleTables() {
-        return tables;
-    }
-
-    @Override
-    public List<CassandraType> moduleTypes() {
-        return ImmutableList.of();
-    }
+public interface CassandraSchemaVersionModule {
+    CassandraModule MODULE = CassandraModule.table(CassandraSchemaVersionTable.TABLE_NAME)
+        .statement(statement -> statement
+            .ifNotExists()
```

Comment:
what about putting this `ifNotExists` into the builder ? It looks more like a policy than a definition, don't you think ?

Reply:
I hesitated about this, but I clearly do agree with you.